src/Microsoft.AspNet.OData.Shared/DeltaOfTStructuralType.cs

```diff
@@ -780,7 +866,10 @@ private void CopyUnchangedDynamicValues(TStructuralType targetEntity)
 
         private bool TrySetPropertyValueInternal(string name, object value)
         {
-            Debug.Assert(name != null, "Argument name is null");
+            if (name == null)
+            {
+                throw Error.ArgumentNull(nameof(name));
+            }
```

Comment:
since it's private, if there's a check the "name" in the caller, we don't need to check it again. Debug.Assert is better here.